app/views/admin/registration_periods/show.html.haml

```diff
@@ -26,4 +26,11 @@
         method: :delete, data: { confirm: 'Are you sure?' }, class: 'btn btn-danger'
     - else
       - if can? :create, @conference.build_registration_period
-        = link_to 'New Registration Period', new_admin_conference_registration_period_path, class: 'btn btn-primary'
+        - unless @conference.registration_tickets.empty?
+          = link_to 'New Registration Period', new_admin_conference_registration_period_path, class: 'btn btn-primary'
```

Comment:
> Although you don't show the link, the path would be still accessible and the admin could create a registration period without registration tickets.

Yes we can create a validation for this in registration period.

> And another concern, what is the admin do not want to make the registration compulsory?

I am assuming if admin wants to set the registration period then it means registration is compulsory for that conference. There might be a case when user does not specifically wants to attend the conference but just want to buy some other kind of ticket like food ticket or something. In this case user can purchase the food ticket but the user won't be registered for the conference and thus cannot be marked as present. So to summarise if admin does not want to make registration compulsory then no registration period is to be set. If admin wants to set registration period then that means registration is compulsory and there must be at least one registration ticket.